common/sampling.h

```diff
@@ -25,6 +25,9 @@ typedef struct llama_sampling_params {
     int32_t     mirostat              = 0;        // 0 = disabled, 1 = mirostat, 2 = mirostat 2.0
     float       mirostat_tau          = 5.00f;    // target entropy
     float       mirostat_eta          = 0.10f;    // learning rate
+    bool        dynatemp              = false;    // dynamic temperature
```

Comment:
So the first thing that stands out first is - I don't think a bool for toggling dynatemp is necessary. I think this can be simplified and there are 2 very good ways of doing it.

First way is to make it part of the sampler, so having `dynatemp_min` and `dynatemp_max` both be 0 would leave dynatemp inactive, otherwise it would be active and applied (overriding temperature). That puts it in line with all other samplers that have an "inactive" value and an "active" value. 

The second option which I am strongly in support is a single value `dynatemp_range`. This is a float which represents the allowed deviation from the min and max temperature when using dynatemp.

Thus, if we want a value of `dynatemp_min=0.3, dynatemp_max=0.5`, then we would simply set `temperature=0.4` and `dynatemp_range=0.1`

If you want a `dynatemp_min=0.8, dynatemp_max=2.0`, then you set `temperature=1.4` and `dynatemp_range=0.6`. 

It will work for any value of min and max

To disable, set `dynatemp_range=0`

Thoughts?

Reply:
I like this idea and it's nicely intuitive.